samcli/commands/_utils/options.py

```diff
@@ -666,6 +683,24 @@ def hook_package_id_click_option(force_prepare=True, invalid_coexist_options=Non
     )
 
 
+def skip_prepare_iac_click_option():
+    """
+    Click option to skip the hook preparation stage
+    """
+    return click.option(
+        "--skip-prepare-iac",
+        is_flag=True,
+        required=False,
+        callback=skip_prepare_iac_callback,
+        help="Skips the preparation stage for the hook package if the metadata file has already been generated. "
```

Comment:
Reading this, makes me question what this magic metadata file is. Do we explain this anywhere in our help texts. if not, we should provide context here.

Reply:
I think we need to rephrase in in a way to let the customers know that they can set this flag if they did not change their infrastructure since last time they use sam commands. The customers actually need to regenerate the metadata file after the change the infrastructure  code

Reply:
New help message is: `This option should be used to skip the preparation stage if there have not been any infrastructure changes. The --hook-name option should also be specified when using this option.`. This is pending UX comments, but this is what is currently proposed.